mongomock/collection.py

```diff
@@ -1359,6 +1367,24 @@ def aggregate(self, pipeline, **kwargs):
             '$millisecond',
             '$dateToString']
 
+        def _handle_aggregate_operator(operator, values, doc_dict):
+            if operator == "$sum":
+                val_it = (_parse_expression(val, doc_dict) or 0 for val in values)
```

Comment:
Actually, I've realized that you parse those expression twice in the case of a group, which is not a good idea. I'll try to factorize but without this problem.